docs/wiki/guides/current/SoftSerial.md

```diff
@@ -0,0 +1,254 @@
+- to get data at higher data rates, eg 10hz, for GPS Rescue and position hold purposes.
+
+For simple position data, if the GPS module has been pre-configured, in uCenter or pyGpsClient, to power up at a low baud rate, eg 19200 baud, or 9600 baud, and to send position data values alone at 1Hz, then Betaflight only needs a single Rx pin to receive that data. This can be a softserial pin. In the GPS tab, auto-configuration _must_ be disabled. The baud rate for the softserial port must be set to match the GPS's power up baud rate. Test first with a real Rx pin and be absolutely certain you are not sending unwanted data, eg satellite position information. You should not expect to see a satellite list in Configurator, only Latitude, Longitude, Altitude, etc. If this works on a real Rx pin, then test it on on a Softserial pin.
+
+When the GPS is intended for GPS Rescue purposes, Betaflight requires bi-directional communication with the GPS module during the initialisation phase, so that we can configure the data rate, request only the data we need, and block any other junk that we don't need. To configure the GPS requires both Rx and Tx lines. Auto-configuration requires moving through all the available baud rates that the GPS may connect on. For most modern GPS modules, the defdault baud rate is 57600, which is too high for a two-wire Softserial setup to work with. As a result the GPS is likely to not get configured properly, and auto-configuration will fail. Even with a manually-configured GPS, with a single Rx pin on the FC for softserial set to 19200 baud and 2Hz, and with GPS auto-configuration disabled, the best you could hope for would be a very jerky GPS Rescue that may fail at any time.. Hence we strongly discourage the use of SoftSerial for GPS Rescue.
+
+### Softserial with HD OSD (displayport) over MSP
+
+This is not recommended, though it _may_ work. Both a SoftserialTx and an Softserial Rx pin must be defined. The data rate with anything but the simplest OSD would likely overwhelm the connection. Note that softserial is limited to 19200 baud so don't use rates above that.
+
+### Enabling Softserial
+
+Softserial requires custom CLI commands. A unused pin with an associated timer must be found, where the original purpose no longer is needed. The most common are PPM pins or LED_STRIP pins. Not all such pins will work on all boards, and you may need to do some reseach about using softserial on your board before you can make it work.
+
+First go to the CLI and type `resource`, to get a list of the currently assigned pin numbers for each of the currently available pads on the device.
+
+If, for instance, a `LED_STRIP 1` pin is defined to `A15`, and there is a LedStrip pad on the board, and you are not using a led strip, you could de-assign that pin from `LED_STRIP 1` with the CLI command:
+
+`resource led_strip 1 none`
+
+Then the pin can be assigned to the Softserial Port 1 Tx pin with:
+
+`resource serialTx 11 A15`
```

Comment:
```suggestion
`resource SOFTSERIAL_TX 1 A15`
```

Serial TX 11 is used for LPUART

Reply:
As far as I am aware, LPUARTs are handled like normal UARTS, ie they will be configured using `serialTx` like any other port,, eg `set serialTx 6  <pin>` like any other UART, in CLI.

Serial TX 11 is not used for LPUART.  It is used to configure a softserial port.  You can confirm this easily in CLI by setting a pin to softserial as described above, then running the serial command, which will show a serial port at number 30.

Reply:
Currently in master `UART 11` is indeed used for `LPUART`.
Softserial has it's own structure using `SOFTSERIAL_[T|R]X`

Reply:
That's something to be discussed elsewhere.  I twas my understanding that LPUARTs are configured using the same syntax as any other UART.

Reply:
That's why I had raised https://github.com/betaflight/betaflight/pull/13278 for more clarity

Reply:
Probably we just leave LPUART as TX/RX11 for 4.5.  I will add some notes.

Reply:
- CLI in firmware does not use the same definition which we use to add LPUART1 in config.
- If manufacturers would add  UART11 then we have a problem.
- LPUART has range 40-49 and should not use UART 1-20 range.
- UART 11 in firmware does not exist. It's called LPUART1

```
configs/AIRBOTG4AIO/config.h:63:#define LPUART1_TX_PIN       PB11
configs/AIRBOTG4AIO/config.h:69:#define LPUART1_RX_PIN       PB10
configs/KAKUTEG4AIO/config.h:54:#define LPUART1_TX_PIN       PB11
configs/KAKUTEG4AIO/config.h:59:#define LPUART1_RX_PIN       PB10
configs/MAMBAG4/config.h:71:#define LPUART1_RX_PIN       PB11
configs/TAKERG4AIO/config.h:50:#define LPUART1_TX_PIN      PB10
configs/TAKERG4AIO/config.h:51:#define LPUART1_RX_PIN      PB11
configs/LUXHDAIO-G4/config.h:51:#define LPUART1_TX_PIN       PB11
configs/LUXHDAIO-G4/config.h:56:#define LPUART1_RX_PIN       PB10
configs/LUXMICROAIO/config.h:56:#define LPUART1_TX_PIN       PB11
configs/LUXMICROAIO/config.h:62:#define LPUART1_RX_PIN       PB10
```

Reply:
Made notes about TX11 being for LPUART1 and explained some stuff about LPUART generally in the release notes also